// @woocommerce/settings/build/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getSetting = getSetting;
exports.DEFAULT_DATE_RANGE = exports.WC_ASSET_URL = exports.SITE_TITLE = exports.ORDER_STATUSES = exports.LOCALE = exports.CURRENCY = exports.COUNTRIES = exports.ADMIN_URL = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _typeof2 = _interopRequireDefault(require("@babel/runtime/helpers/typeof"));

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(source, true).forEach(function (key) { (0, _defineProperty2.default)(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(source).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var defaults = {
  adminUrl: '',
  countries: [],
  currency: {
    code: 'USD',
    precision: 2,
    symbol: '$',
    symbolPosition: 'left',
    decimalSeparator: '.',
    priceFormat: '%1$s%2$s',
    thousandSeparator: ','
  },
  defaultDateRange: 'period=month&compare=previous_year',
  locale: {
    siteLocale: 'en_US',
    userLocale: 'en_US',
    weekdaysShort: ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat']
  },
  orderStatuses: [],
  siteTitle: '',
  wcAssetUrl: ''
};
var globalSharedSettings = (typeof wcSharedSettings === "undefined" ? "undefined" : (0, _typeof2.default)(wcSharedSettings)) === 'object' ? wcSharedSettings : {}; // Use defaults or global settings, depending on what is set.

var allSettings = _objectSpread({}, defaults, {}, globalSharedSettings);

allSettings.currency = _objectSpread({}, defaults.currency, {}, allSettings.currency);
allSettings.locale = _objectSpread({}, defaults.locale, {}, allSettings.locale);
var ADMIN_URL = allSettings.adminUrl;
exports.ADMIN_URL = ADMIN_URL;
var COUNTRIES = allSettings.countries;
exports.COUNTRIES = COUNTRIES;
var CURRENCY = allSettings.currency;
exports.CURRENCY = CURRENCY;
var LOCALE = allSettings.locale;
exports.LOCALE = LOCALE;
var ORDER_STATUSES = allSettings.orderStatuses;
exports.ORDER_STATUSES = ORDER_STATUSES;
var SITE_TITLE = allSettings.siteTitle;
exports.SITE_TITLE = SITE_TITLE;
var WC_ASSET_URL = allSettings.wcAssetUrl;
exports.WC_ASSET_URL = WC_ASSET_URL;
var DEFAULT_DATE_RANGE = allSettings.defaultDateRange;
exports.DEFAULT_DATE_RANGE = DEFAULT_DATE_RANGE;

function getSetting(name) {
  var fallback = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : false;

  if (allSettings.hasOwnProperty(name)) {
    return allSettings[name];
  }

  return fallback;
}
